### ISSUE TITLE
lolcate --update does not handle dangling symbolic links gracefully

### ISSUE THREAD

**Issue by BartMassey** [ngirard/lolcate-rs#19]
When trying to update the database, if a target directory contains a dangling symlink an error message like the following is printed:

```
Updating /tmp/example...
failed to access entry (/tmp/dangling/nonexistent: IO error for operation on /tmp/dangling/nonexistent: No such file or directory (os error 2))
```

(This is from indexing `/tmp/dangling` containing symlink `nonexistent` → `/nope`)

Built from current master commit f876603, Debian Linux

**Comment by BartMassey** [comment]
Forgot to say: this happens regardless of the setting of `ignore_symlinks`.

**Comment by BartMassey** [comment]
This seems to fix things such that the setting of `ignore_symlinks` is respected. The error when symlinks are being followed is expected. I won't bother with a pull request for this unless you really want me to.

```diff
diff --git a/src/main.rs b/src/main.rs
index cee9be6..4751de8 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -290,7 +290,7 @@ pub fn walker(config: &config::Config, database: &st
r) -> ignore::Walk {
     let mut wd = ignore::WalkBuilder::new(&paths[0]);
     wd.hidden(config.ignore_hidden) // Whether to ignore hidden files
         .parents(false) // Don't read ignore files from parent directories
-        .follow_links(true) // Follow symbolic links
+        .follow_links(!config.ignore_symlinks) // Follow symbolic lin
ks
         .ignore(true) // Don't read .ignore files
         .git_global(config.gitignore) // Don't read global gitignore file
         .git_ignore(config.gitignore) // Don't read .gitignore files
```

**Comment by ngirard** [comment]
Good catch, thanks ! No need for a PR here, I'll make sure to credit you in the CHANGELOG.

**Comment by ngirard** [comment]
> This seems to fix things such that the setting of `ignore_symlinks` is respected.

Just pushed a [new release](https://github.com/ngirard/lolcate-rs/releases/tag/v0.6.1) with the fix.

**Comment by ngirard** [comment]
I'm closing this issue as it's supposed to be fixed since v0.6.1.
@BartMassey, feel free to reopen it if necessary.
